analyses/survival-analysis/survival-analysis_histologies.Rmd

```diff
@@ -0,0 +1,236 @@
+         telomerase_score = NormEXTENDScores) %>%
+  select(Kids_First_Biospecimen_ID_RNA, telomerase_score)
+
+# remove samples without RNA-Seq since both classifiers' results use RNA-Seq
+# independent use primary plus rna-seq samples
+meta_indep <- tp53_scores %>%
+  left_join(tel_scores) %>%
+  left_join(histologies) %>%
+  dplyr::filter(Kids_First_Biospecimen_ID_RNA %in% indep_rna$Kids_First_Biospecimen_ID)
+
+```
+
+Caclulte the quantiles for both tp53 scores and telomarase scores. 
+These scores are used later on for classifying samples into different categories for survival analysis.
+```{r}
+meta_indep_no_hgat <- meta_indep %>%
+  filter(short_histology != "HGAT")
+
+tp53_quantiles <- quantile(meta_indep$tp53_score, probs = seq(0, 1, 0.25))
+telomerase_quantiles<- quantile(meta_indep$telomerase_score, probs = seq(0, 1, 0.25))
+
+tp53_quantiles_no_hgat <- quantile(meta_indep_no_hgat$tp53_score, probs = seq(0, 1, 0.25))
+telomerase_quantiles_no_hgat <- quantile(meta_indep_no_hgat$telomerase_score, probs = seq(0, 1, 0.25))
+```
+
+Create groupings for scores - basically, based on the TP53 and telomarase activity scores, additional columns are added indicating which group a particular sample fits in:
+- `tp53_0.5`: this columns indicates whether a sample is `tp53_high` or `tp53_low` based on whehter its TP53 score is over or below 0.5
+- `tel_0.5`: this columns indicates whether a sample is `telomerase_high` or `telomerase_low` based on whehter its telomerase score is over or below 0.5
+- `tp53_strict`: this columns indicates whether a sample is `tp53_high`, `tp53_mid`, or `tp53_low` based on whether they are above 75 quantile (`tp53_high`), below 25 quantile (`tp53_low`), or in between (`tp53_mid`)
+- `tel_strict`: this columns indicates whether a sample is `telomerase_high`, `telomerase_mid`, or `telomerase_low` based on whether they are above 75 quantile (`telomerase_high`), below 25 quantile (`telomerase_low`), or in between (`telomerase_mid`)
+- `pheno_0.5`: this column combines info from `tp53_0.5` and `tel_0.5`
+- `pheno_strict`: this column combines info from `tp53_strict` and `tel_strict`
+- `pheno_extremes`: this column only retains categorizations from `pheno_strict` column when the sample does not fit into `mid` group (between 25 and 75 percentile) for neither TP53 nor telomase scores
+
+```{r}
+meta_indep <- calc_groups(meta_indep, tp53_quantiles, telomerase_quantiles)
+meta_indep_no_hgat <- calc_groups(meta_indep_no_hgat, tp53_quantiles_no_hgat, telomerase_quantiles_no_hgat)
+```
+
+Set phenotypes to loop through. These are columns created by the `calc_groups` function. These columns, as described above, contain information about whether the sample has `high`, `mid` or `low` TP53/telomerase scores.
+```{r}
+# phenotypes <- c("tp53_0.5", "tel_0.5", "tp53_strict", "tel_strict", "pheno_0.5", "pheno_strict", "pheno_extremes")
+phenotypes <- c("tp53_0.5", "tel_0.5", "tp53_strict", "tel_strict", "pheno_0.5", "pheno_strict")
```

Comment:
I removed `pheno_extremes` here to (hopefully) pass CI, but I wonder if we really want it at all, as I discussed in my earlier comment.